lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -2570,4 +2575,71 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
       });
+
+  patterns.onOp(
+      "DFT", 20, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Value inTensor, dftLength, axis;
+        Torch::ValueTensorType resultType;
+        int64_t inverse, onesided;
+        if (binder.tensorOperandAtIndex(inTensor, 0))
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Input Tensor bind failed");
+        if (!binder.tensorOperandAtIndex(dftLength, 1)) {
+          // Convert to int and pass as n
+          dftLength = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), dftLength);
+        } else {
+          // Default for torch is None
+          dftLength = rewriter.create<Torch::ConstantNoneOp>(binder.getLoc());
+        }
+        // Default is same for onnx and torch
+        if (!binder.tensorOperandAtIndex(axis, 2)) {
+          // convert to int and pass to dims
+          axis = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), axis);
+        } else {
+          // Default in torch is -1 and onnx is -2 (since -1 is for real / img)
+          axis = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(-2));
+        }
+        if (binder.s64IntegerAttr(inverse, "inverse", 0) ||
+            binder.s64IntegerAttr(onesided, "onesided", 0) ||
+            binder.tensorResultType(resultType)) {
+          return rewriter.notifyMatchFailure(
+              binder.op, "bind failed for attrs / resultType");
+        }
+
+        if (onesided == 1)
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Unsupported option : onesided");
+        // norm default string attr
+        Value norm = rewriter.create<Torch::ConstantStrOp>(
+            binder.getLoc(), rewriter.getStringAttr(Twine("backward")));
+        inTensor.getType().dump();
+        Torch::ValueTensorType inType =
+            binder.toValidTensorType(inTensor.getType());
+        inType.getDtype().dump();
+        Value inComplexTensor = rewriter.create<Torch::AtenViewAsComplexOp>(
+            binder.getLoc(),
+            Torch::ValueTensorType::get(
+                binder.op->getContext(), inType.getSizes(),
+                mlir::ComplexType::get(inType.getDtype())),
+            inTensor);
+        Value ftOp;
+        if (inverse == 0) {
+          ftOp = rewriter.create<Torch::AtenFftFftOp>(
+              binder.getLoc(),
+              Torch::ValueTensorType::get(
+                  binder.op->getContext(), resultType.getSizes(),
+                  mlir::ComplexType::get(inType.getDtype())),
+              inComplexTensor, /*n = */ dftLength, /*dim = */ axis,
+              /*norm = */ norm);
+        } else {
+          return rewriter.notifyMatchFailure(binder.op,
+                                             "Inverse FFT unimplemented");
```

Comment:
You can add the support for inverse as well by lowering it to https://pytorch.org/docs/stable/generated/torch.fft.ifft.html.